v0/destinations/canny/transform.js

```diff
@@ -0,0 +1,174 @@
+  contentType,
+  responseBody
+) => {
+  const response = defaultRequestConfig();
+  if (payload) {
+    response.endpoint = `${BASE_URL}${endpoint}`;
+    response.headers = {
+      Authorization: `Basic ${apiKey}`,
+      "Content-Type": contentType
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body[`${responseBody}`] = removeUndefinedAndNullValues(payload);
+  }
+  return response;
+};
+
+const identifyResponseBuilder = async (message, { Config }) => {
+  const { apiKey } = Config;
+  const contentType = "application/json";
+  const responseBody = "JSON";
+
+  if (!apiKey) {
+    throw new CustomError("API Key is not present. Aborting message.", 400);
+  }
+
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategory.IDENTIFY.name]
+  );
+
+  if (!payload.userID) {
+    throw new CustomError("UserId is not present. Aborting message.", 400);
+  }
+  if (!payload.name) {
+    throw new CustomError("Name is not present. Aborting message.", 400);
+  }
+
+  const { endpoint } = ConfigCategory.IDENTIFY;
+
+  return responseBuilder(payload, apiKey, endpoint, contentType, responseBody);
+};
+
+const getTrackResponse = async (apiKey, message, val) => {
+  let endpoint;
+  let responseBody;
+  let contentType;
+  let payload;
+  if (val === "createVote") {
+    responseBody = "FORM";
+    contentType = "application/x-www-form-urlencoded";
+
+    payload = constructPayload(
+      message,
+      mappingConfig[ConfigCategory.CREATE_VOTE.name]
+    );
+    if (!payload.postID) {
+      throw new CustomError("PostID is not present. Aborting message.", 400);
+    }
+
+    payload.apiKey = apiKey;
+    payload.voterID = "voterid";
+    const voterID = await retrieveUserId(apiKey, message);
+    payload.voterID = voterID;
+    endpoint = ConfigCategory.CREATE_VOTE.endpoint;
+  } else if (val === "createPost") {
+    contentType = "application/json";
+    responseBody = "JSON";
+
+    payload = constructPayload(
+      message,
+      mappingConfig[ConfigCategory.CREATE_POST.name]
+    );
+
+    if (!payload.boardID) {
+      throw new CustomError("BoardID is not present. Aborting message.", 400);
+    }
+    if (!payload.title) {
+      throw new CustomError("Title is not present. Aborting message.", 400);
+    }
+    if (!payload.details) {
+      throw new CustomError("Details is not present. Aborting message.", 400);
+    }
+
+    // payload.authorID = "authorid";
+    payload.authorID = await retrieveUserId(apiKey, message);
+
+    endpoint = ConfigCategory.CREATE_POST.endpoint;
+  }
+
+  return responseBuilder(payload, apiKey, endpoint, contentType, responseBody);
+};
+
+const trackResponseBuilder = async (message, { Config }) => {
+  const { apiKey } = Config;
+  const eventsToEvents = Config.eventsToEvents;
+  const eventsMap = getHashFromArrayWithDuplicate(eventsToEvents);
+
+  if (!apiKey) {
+    throw new CustomError("API Key is not present. Aborting message.", 400);
+  }
```

Comment:
This is repeating put them inside process function.

Reply:
Done